ann_benchmarks/algorithms/luceneknn.py

```diff
@@ -0,0 +1,124 @@
+"""
+ann-benchmarks interface for Apache Lucene.
+"""
+
+import sklearn.preprocessing
+import numpy as np
+from multiprocessing.pool import ThreadPool
+
+
+import lucene
+from lucene import JArray
+from java.nio.file import Paths
+from org.apache.lucene.store import FSDirectory
+from org.apache.lucene.search import KnnVectorQuery, IndexSearcher
+from org.apache.lucene.index import IndexWriter, IndexWriterConfig, VectorSimilarityFunction, DirectoryReader
+from org.apache.lucene.codecs.lucene94 import Lucene94Codec, Lucene94HnswVectorsFormat
+from org.apache.lucene.document import Document, FieldType, KnnVectorField, StoredField
+
+from ann_benchmarks.algorithms.base import BaseANN
+
+
+class Codec(Lucene94Codec):
+    """
+    Custom codec so that the appropriate Lucene94 codec can be returned with the configured M and efConstruction
+    """
+    def __init__(self, M, efConstruction):
+        super(Codec, self).__init__()
+        self.M = M
+        self.efConstruction = efConstruction
+
+    def getKnnVectorsFormatForField(self):
+        Lucene94HnswVectorsFormat(self.M, self.efConstruction)
```

Comment:
I think we need to explicitly `return` here, otherwise there's no return value and Lucene will always use the default parameters for M and efConstruction. (Maybe this is why we were seeing unexpected recall values in our tests?)

Reply:
Literal LOL!